dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/EventImporterValidationRunner.java

```diff
@@ -27,54 +27,63 @@
  */
 package org.hisp.dhis.dxf2.events.importer;
 
-import static org.apache.commons.logging.LogFactory.getLog;
+import static org.hisp.dhis.dxf2.importsummary.ImportStatus.ERROR;
+import static org.hisp.dhis.dxf2.importsummary.ImportStatus.WARNING;
 
+import java.util.ArrayList;
 import java.util.List;
 
-import org.apache.commons.logging.Log;
+import lombok.RequiredArgsConstructor;
+
 import org.hisp.dhis.dxf2.events.event.Event;
 import org.hisp.dhis.dxf2.events.importer.context.WorkContext;
+import org.hisp.dhis.dxf2.events.importer.shared.ImmutableEvent;
+import org.hisp.dhis.dxf2.importsummary.ImportSummary;
+import org.springframework.stereotype.Component;
 
-/**
- * Simple interface that provides processing capabilities on events.
- *
- * @author maikel arabori
- */
-public interface EventProcessing
+@Component
+public class EventImporterValidationRunner
 {
-    void process( WorkContext workContext, List<Event> events );
-
-    Log log = getLog( EventProcessing.class );
-
-    class ProcessorRunner
+    /**
+     * Validates the events using the supplied list of validators.
+     * <p>
+     * Only returns the ImportSummary for Events that *did* not pass validation
+     *
+     * @param payload Validation Runner payload to run against
+     * @return returns the ImportSummary for Events that did not pass validation
+     */
+    public List<ImportSummary> run( ValidationRunnerPayload payload )
     {
-        private final WorkContext workContext;
-
-        private final List<Event> events;
-
-        public ProcessorRunner( WorkContext workContext, List<Event> events )
-        {
-            this.workContext = workContext;
-            this.events = events;
-        }
+        final List<ImportSummary> importSummaries = new ArrayList<>( 0 );
```

Comment:
Initializing this with `0` strikes me as odd.
Do we expect that a successful run has zero summaries?

Reply:
I didn't notice that, it was existing code, I'll change it